### ISSUE TITLE
[plugin] sosreport does not collect symlinks targets

### ISSUE THREAD

**Issue by pmoravec** [sosreport/sos#618]
Reproducer:

```
touch /etc/empty
ln -s /etc/empty /etc/pam.d/emtpy
sosreport --batch -o pam -e pam
```

in unpacked sosreport archive:

```
$ file etc/pam.d/empty 
etc/pam.d/empty: broken symbolic link to `../empty'
$
```

The root cause is `self._do_copy_path` in `_copy_symlink` calls `strip_sysroot` everytime. This can be fixed by:

```
diff --git a/sos/plugins/__init__.py b/sos/plugins/__init__.py
index aed7496..601264a 100644
--- a/sos/plugins/__init__.py
+++ b/sos/plugins/__init__.py
@@ -297,7 +297,7 @@ class Plugin(object):

         # skip recursive copying of symlink pointing to itself.
         if (absdest != srcpath):
-            self._do_copy_path(self.strip_sysroot(absdest))
+            self._do_copy_path(self.strip_sysroot(absdest) if self.use_sysroot() else absdest)
         else:
             self._log_debug("link '%s' points to itself, skipping target..."
                             % linkdest)
```

But:
- the patch hasn't been tested in a container and might have other side-effects I dont see now from my narrow point of view
- I would rather add `use_sysroot()` check into `strip_sysroot` function to prevent those gotchas next time

**Comment by bmr-cymru** [comment]
That check is supposed to be in there; I thought it had been pushed to bmr-tupperware but looking there I don't see it (it's possible in the rush in Jan/Feb I put it straight into the RPM build in which case me--). Anyway, this is what the fix should look like:

```python
    def strip_sysroot(self, path):
        if not self.use_sysroot():
            return path
        if path.startswith(self.sysroot):
            return path[len(self.sysroot):]
        return path
```

There's no need for an equivalent in the `join_sysroot()` method; `os.path.join`ing  `/` to an abs path is a no-op.

**Comment by bmr-cymru** [comment]
Fixed many years ago in commit 4d8b5418b1cf218ef9746674ca5935a7f4f3983e.
